contrib/gin-gonic/gin/gintrace_test.go

```diff
@@ -340,6 +341,54 @@ func TestResourceNamerSettings(t *testing.T) {
 	})
 }
 
+func TestIgnoreRequestSettings(t *testing.T) {
+	tests := []struct {
+		url       string
+		spanCount int
+	}{
+		{
+			url:       "/skip",
+			spanCount: 0,
+		},
+		{
+			url:       "/skipfoo",
+			spanCount: 0,
+		},
+		{
+			url:       "/OK",
+			spanCount: 1,
+		},
+	}
+
+	router := gin.New()
+	router.Use(Middleware("foobar", WithIgnoreRequest(func(c *gin.Context) bool {
+		return strings.HasPrefix(c.Request.URL.Path, "/skip")
+	})))
+
+	router.GET("/OK", func(c *gin.Context) {
+		c.Writer.Write([]byte("OK"))
+	})
+
+	router.GET("/skip", func(c *gin.Context) {
+		c.Writer.Write([]byte("Skip"))
+	})
+
+	for _, test := range tests {
```

Comment:
I would prefer it if we didn't create the `tests` variable. There is no need for it, it's not used in more than one place. You can just inline all the tests. If you like (up to you), it could also be a map:
```suggestion
	for path, count := range map[string]int{
		"/OK": 1,
		// ...
	 } {
```

Reply:
+1 on keeping the tests variable. Easier to read(table driven tests), can also add a field called `expectedToSkip bool` and match on this later.

Reply:
Perhaps you can meet both the suggested requirements by doing something like this:
```go
for path, shouldSkip := range map[string]bool{
    "/OK": false,
    // ...
} {
   // ...
}
```

Reply:
Sounds good, made the change 👍